lua/lvim/core/breadcrumbs.lua

```diff
@@ -99,33 +99,40 @@ M.get_filename = function()
   local f = require "lvim.utils.functions"
 
   if not f.isempty(filename) then
-    local file_icon, hl_group = require("nvim-web-devicons").get_icon(filename, extension, { default = true })
+    local file_icon, hl_group
+    local devicons_ok, devicons = pcall(require, "nvim-web-devicons")
+    if lvim.use_icons and devicons_ok then
+      file_icon, hl_group = devicons.get_icon(filename, extension, { default = true })
 
-    if f.isempty(file_icon) then
-      file_icon = lvim.icons.kind.File
-    end
+      if f.isempty(file_icon) then
+        file_icon = lvim.icons.kind.File
+      end
 
-    local buf_ft = vim.bo.filetype
+      local buf_ft = vim.bo.filetype
 
-    if buf_ft == "dapui_breakpoints" then
-      file_icon = lvim.icons.ui.Bug
-    end
+      if buf_ft == "dapui_breakpoints" then
+        file_icon = lvim.icons.ui.Bug
+      end
 
-    if buf_ft == "dapui_stacks" then
-      file_icon = lvim.icons.ui.Stacks
-    end
+      if buf_ft == "dapui_stacks" then
+        file_icon = lvim.icons.ui.Stacks
+      end
 
-    if buf_ft == "dapui_scopes" then
-      file_icon = lvim.icons.ui.Scopes
-    end
+      if buf_ft == "dapui_scopes" then
+        file_icon = lvim.icons.ui.Scopes
+      end
 
-    if buf_ft == "dapui_watches" then
-      file_icon = lvim.icons.ui.Watches
-    end
+      if buf_ft == "dapui_watches" then
+        file_icon = lvim.icons.ui.Watches
+      end
```

Comment:
these don't depend on the package being available or not, so you should not nest the condition.

alternatively, you could return early if `lvim.icons` is turned off.